pynetbox_data_uploader/lib/utils/csv_to_dataclass.py

```diff
@@ -0,0 +1,41 @@
+import csv
+from typing import List
+from lib.utils.dataclass_data import Device
+
+
+def open_file(file_path: str) -> csv.DictReader:
+    """
+    This function opens the specified csv file and returns the DictReader class on it.
+    :param file_path: The file path to the csv file.
+    :return: Returns an instance of the DictReader Class.
+    """
+    with open(file_path, encoding="UTF-8") as file:
+        csv_reader_obj = csv.DictReader(file)
+    return csv_reader_obj
+
+
+def separate_data(csv_reader_obj: csv.DictReader) -> List[Device]:
+    """
+    This method separates the data from the iterator object into a list of dataclasses for each device.
+    :param csv_reader_obj: The DictReader class with the data.
+    :return: Returns a list of dataclass objects.
+    """
+    devices = []
+    for row in csv_reader_obj:
+        device = Device(
+            tenant=row["tenant"],
+            device_role=row["device_role"],
+            manufacturer=row["manufacturer"],
+            device_type=row["device_type"],
+            status=row["status"],
+            site=row["site"],
+            location=row["location"],
+            rack=row["rack"],
+            face=row["face"],
+            airflow=row["airflow"],
+            position=row["position"],
+            name=row["name"],
+            serial=row["serial"],
+        )

```

Comment:
This is fine - but you could create a method called `from_dict` in the dataclass to make this a bit tidier 
```
from dataclasses import dataclass, fields

from_dict(device_dict: Dict):
        field_set = {field.name for field in fields(Device) if field.init}
        filtered_arg_dict = {
            key: value for key, value in device_dict.items() if key in field_set
        }
        return Device(**filtered_arg_dict)
```

Reply:
Will consider that in future PR.